broker-core/src/main/java/io/zeebe/broker/engine/impl/DeploymentDistributorImpl.java

```diff
@@ -237,4 +299,15 @@ private void handlePushResponse() {
           pushDeploymentResponse.partitionId());
     }
   }
+
+  public static String getDeploymentResponseTopic(final long deploymentKey) {
+    return String.format("deployment-response-%d", deploymentKey);
+  }
+
+  private Int2IntHashMap getPartitionResponses(final String topic) {
+    final Int2IntHashMap responses =
+        deploymentResponses.getOrDefault(topic, new Int2IntHashMap(NO_RESPONSE));
```

Comment:
It seems a bit weird with the `NO_RESPONSE` and `RESPONSE` constants, which could be btw also a enum.
Why not using the same logic where you have the list of partition ids and remove the id from the list from which you get the response. In the end the list should be empty and you now you can write the distributed event.

Reply:
Actually I also not seeing where you really use this values at all?